ecs-agent/acs/session/attach_resource_responder_test.go

```diff
@@ -244,6 +248,23 @@ func testValidateAttachmentAndReturnPropertiesWithAttachmentType(t *testing.T) {
 		}
 		require.True(t, verified, "Missing required property: %s", requiredProperty)
 	}
+
+	for _, property := range confirmAttachmentMessageCopy.Attachment.AttachmentProperties {
+		if aws.StringValue(property.Name) == resource.FileSystemKey {
+			originalPropertyValue := property.Value
+			property.Value = aws.String("SomeFilesystemType")
+			_, err = validateAttachmentAndReturnProperties(&confirmAttachmentMessageCopy)
+			require.Error(t, err)
+			property.Value = originalPropertyValue
+
+			originalPropertyValue = property.Value
+			property.Value = aws.String("")
+			_, err = validateAttachmentAndReturnProperties(&confirmAttachmentMessageCopy)
+			require.Error(t, err)
+			property.Value = originalPropertyValue
+		}
+	}
```

Comment:
In future I recommend separating out test cases from test runner code to better readability.

Reply:
yep I agree, the unit test in our attach resource responder is a bit messy to read. I'll add a todo

Reply:
actually, I like the idea of separating this out of the runner code. I made it's own test

Reply:
thanks!